### PR TITLE
feat(chat-client): add Web UI

### PR DESCRIPTION
**Pull request by nickvidal**
Add Web UI for chat clients.

Waiting for PR #70 to get merged before making connections from Rust, C++, and C chat clients.

Signed-off-by: Nick Vidal <nick@profian.com>

### CODE REVIEW

File: demos/chat-client/chat-web-ui/script.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    chatPopup = document.querySelector(".chat-popup")
+    chatBtn = document.querySelector(".chat-btn")
+    chatSubmit = document.querySelector(".chat-submit")
+    chatHeader = document.querySelector(".chat-header")
+    chatArea = document.querySelector(".chat-area")
+    chatInput = document.querySelector(".chat-input")
+    expandWindow = document.querySelector(".expand-chat-window")
+    root = document.documentElement;
+    chatPopup.style.display = "none"
+    var host = "http://localhost:8080"
```
Comment (rvolosatovs) [inline demos/chat-client/chat-web-ui/script.js:14]: This should:
1. Use something more unique. I suggested port `50010` in #69 
2. Use `https`

File: demos/chat-client/chat-web-ui/script.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+function createChatBot() {
+
+    host = "http://localhost:8080";
+    init()
+    const msg = document.querySelector(".msg");
+    msg.innerText = "Welcome to Enarx Chat!";
+
+    const botTitle = document.querySelector(".bot-title");
+    botTitle.innerText = "Enarx Chat";
+}
```
Comment (rvolosatovs) [inline demos/chat-client/chat-web-ui/script.js:125]: There should be no bots, if you want to initialize elements with some data, just do that directly in initialization routine

File: demos/chat-client/chat-web-ui/script.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    }
+}
+
+function detectMob() {
+    return ((window.innerHeight <= 800) && (window.innerWidth <= 600));
+}
+
+function createChatBot() {
+
+    host = "http://localhost:8080";
```
Comment (rvolosatovs) [inline demos/chat-client/chat-web-ui/script.js:118]: unused?

**Review by rvolosatovs — CHANGES_REQUESTED** [review]
Let's move
`/demos/chat-client/chat-web-ui` -> `/demos/chat-client/web-ui`

There should not be a pop-up, rather once opened, the maximized chat window should immediately appear without having to click any buttons.

I only see functionality to send messages, how are messages ever fetched from the server?
In any case, this UI should rely on a WebSocket to send and receive messages https://developer.mozilla.org/en-US/docs/Web/API/WebSocket

**Comment by nickvidal** [comment]
The Chat window is now fully expanded. Maybe one of the students can work on the WebSocket?

**Review by rvolosatovs — CHANGES_REQUESTED** [review]
@nickvidal this still does not fit the purpose.

We should be printing lines of text to a blank page and be able to send lines of text.
There is no concept of a "user" or anything like that in the protocol

I would expect to see a white page with black text on top, printed directly from the server and a small form below to send a line.

There cannot be a bot here.

Why is not just all the text printed on the page itself? What's the purpose of the "box"?

**Comment by rvolosatovs** [comment]
I haven't seen @kieranjazy's UI, but I have a feeling that that work was much better suited for what we're building. Where can I find that?

**Comment by rvolosatovs** [comment]
These messages are *not* meant for humans.
We may come up with something humans can interact with, but that will not be uncensored text.

Note that even if we wanted to build a generic chat application for humans, we simply could not publish an anonymous, uncensored chat on https://try.enarx.dev

**Review by rvolosatovs — APPROVED** [review]
Let's handle the rest in a follow-up

**Pull request merged**